packages/pluggableWidgets/rich-text-web/src/utils/ckeditorConfigs.ts

```diff
@@ -0,0 +1,82 @@
+import { CKEditorConfig } from "ckeditor4-react";
+import { PresetEnum, AdvancedConfigType, CtItemTypeEnum } from "../../typings/RichTextProps";
+import { createPreset, TOOLBAR_GROUP, ToolbarGroup, ToolbarItems } from "./ckeditorPresets";
+
+export type PluginName = "codesnippet";
+
+const PLUGIN_CONFIGS = {
+    codesnippet: {
+        extraPlugins: "codesnippet",
+        name: "CodeSnippet",
+        config: {
+            codeSnippet_theme: "idea"
+        }
+    }
+};
+
+export function getToolbarGroupByName(name: string): ToolbarGroup | undefined {
+    return TOOLBAR_GROUP.find((group: ToolbarGroup) => group.name === name);
+}
+
+export function defineEnterMode(type: string): number {
+    switch (type) {
+        case "paragraph":
+            return 1;
+        case "breakLines":
+            return 2;
+        case "blocks":
+            return 3;
+        default:
+            return 1;
+    }
+}
+
+export function getPreset(type: PresetEnum): CKEditorConfig {
+    switch (type) {
+        case "standard":
+            return createPreset("standard");
+        case "basic":
+            return createPreset("basic");
+        case "full":
+            return createPreset("full");
+        default:
+            return createPreset("basic");
+    }
+}
+
+export function addPlugin(name: PluginName, config: CKEditorConfig): CKEditorConfig {
+    const plugin = PLUGIN_CONFIGS[name];
+    if (plugin && config) {
+        if (config.extraPlugins && !config.extraPlugins.includes(plugin.extraPlugins)) {
+            config.extraPlugins += `,${plugin.extraPlugins}`;
+        } else {
+            config.extraPlugins = plugin.extraPlugins;
+        }
+    }
+    return config;
+}
+
+export function defineAdvancedGroups(items: AdvancedConfigType[]): ToolbarItems[] {
+    const toolbarObj: {
+        [key: string]: Array<CtItemTypeEnum | "-">;
+    } = {};
+    items.forEach(item => {
+        const id = item.ctItemToolbar;
+        const type = item.ctItemType !== "seperator" ? item.ctItemType : "-";
+
+        if (!toolbarObj[id]) {
+            toolbarObj[id] = [];
+        }
+        toolbarObj[id].push(type);
+    });
+
+    const keys: string[] = Object.keys(toolbarObj);
+    const toolbarArray: ToolbarItems[] = [];
+    keys.forEach((key: string) =>
+        toolbarArray.push({
+            name: key,
+            items: toolbarObj[key]
+        })
+    );
+    return toolbarArray;
```

Comment:
I have the feeling all lines above can be combined in one map

Reply:
We can't combine it in one map, maybe nested

Input example
```
[
    {ctItemToolbar: "toolbar-1", ctItemType: "separator"},
    {ctItemToolbar: "toolbar-1", ctItemType: "About"},
    {ctItemToolbar: "toolbar-2", ctItemType: "Document"}
]
```
Desired output
```
[
    {name: "toolbar-1", items: ["About", "-"]},
    {name: "toolbar-2", items: ["Document"]}
]
```

Reply:
Although I think it's possible, I don't see a quick way to get rid of the for loop from L60 to L71. But for Line 73 until 81, you can do:
```
return Object.entries(toolbarObj).map(([key, value]) => ({ name: key, items: value }));
```

Reply:
So now you want me to replace 73-81 with this because of what? Just to get rid of `forEach`?